[skip ci] Create Installing_Troubleshooting_Galaxy_WH_6U.md, test/debug guide fo…

bbeggsTT opened the pull request:
### Ticket
N/A

### Problem description
Creating larger user guide for Galaxy WH, this guide needs specific install/debug/test section for 6U systems.

### What's changed
Creating new document for test/debug procedures.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

bbeggsTT committed:
[skip ci] Create Troubleshooting_Galaxy_WH_6U.md, test/debug guide for Galaxy WH 6U systems

Creating a test/debug guide for Galaxy WH 6U systems.

tt-aho reviewed (changes requested):
It is not clear to me that this file location is the appropriate place for this document.
This folder just contains a cmake file, and how would users even know to go to this random folder location? @tt-rkim @bkeith-TT  any ideas where something like this should go?

I would also suggest in the future as a sanity test to just try running any commands listed to make sure they actually work if possible. It would not be great to give users incorrect commands.

tt-rkim reviewed (changes requested):
+1 for Austin's comments

Why is this https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/1074659406/TG+6U+Metal+infra+Galaxy+notes+FAQ

not sufficient?

Do we want this to be public information?

bbeggsTT committed:
Update Troubleshooting_Galaxy_WH_6U.md, edits per comments by Austin and Raymond

minor edits

bbeggsTT committed:
Update Troubleshooting_Galaxy_WH_6U.md

bbeggsTT commented:
I'm fine re-locating this document to the best location. Would this fit better in the tech reports section?

tt-rkim commented:
I can't access that. Please fix the permissions so we can see.

I see the value in adding such, but please review what I posted, as it's literally the same stuff. We have created this very same list and posted it multiple times in the past, and I don't see a good reason to do it again in the repo in lieu of what I posted. I'm still waiting an answer to my question about the doc /upstream tests I posted in my previous comment.

bbeggsTT commented on tt_metal/test/Troubleshooting_Galaxy_WH_6U.md:
```diff
@@ -0,0 +1,25 @@
+- `TT_METAL_SKIP_EHT_CORES_WITH_RETRAIN=1 ./build/test/tt_metal/unit_tests_dispatch --gtest_filter="CommandQueueSingleCardFixture.*"` - This test ensures that basic Command Queue APIs function correctly.
+- `TT_METAL_SKIP_ETH_CORES_WITH_RETRAIN=1 ./build/test/tt_metal/unit_tests_dispatch --gtest_filter="CommandQueueSingleCardProgramFixture.*"` - This test ensures that Metal Program APIs function correctly.
+- 'TT_METAL_SKIP_ETH_CORES_WITH_RETRAIN=1 ./build/test/tt_metal/unit_tests_dispatch --gtest_filetr="CommandQueueSingleCardBufferFixture.ShardedBufferLarge*ReadWrites" - This test ensures that basic Metal Buffers read and write to both L1 and DRAM memory buffers.
+- `./build/test/tt_metal/tt_fabric/fabric_unit_tests --gtest_filter="Fabric2D*Fixture.*"` - This test sets up the Galaxy as a 2D mesh and uses basic Fabric APIs to read and write between chips.
```
Since I created this document I would be fine taking responsibility updating the document.

bbeggsTT commented:
Since this document and the one you linked serve the same purpose would it make sense to combine them? Do we think the SharePoint or the GitHub location has better visibility? We are currently verifying that those tests do indeed work. I am open to open to combining both these documents and posting to the location (sharepoint or github) where more people can see it. Let me know what you think.

bbeggsTT committed:
Update Troubleshooting_Galaxy_WH_6U.md

bbeggsTT committed:
Update and rename Troubleshooting_Galaxy_WH_6U.md to Installing_Troubleshooting_Galaxy_WH_6U.md

Added Installation steps and FAQ section. All installation and testing steps have been verified.

bbeggsTT committed:
Update Installing_Troubleshooting_Galaxy_WH_6U.md, images added

images added

bbeggsTT commented:
All tests and commands have been verified. Added Galaxy Wormhole installation steps. After speaking with @dsklavos I added the FAQ section from @tt-rkim's doc: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/1074659406/TG+6U+Metal+infra+Galaxy+notes+FAQ
As @tt-aho mentioned I think this document will fit better in a different location, possibly the tech reports section.

bbeggsTT committed:
Rename tt_metal/test/Installing_Troubleshooting_Galaxy_WH_6U.md to docs/source/tech_reports/WH_Galaxy/Installing_Troubleshooting_Galaxy_WH_6U.md

location updated

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1

bbeggsTT committed:
Update Installing_Troubleshooting_Galaxy_WH_6U.md

typo fixed

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1

bbeggsTT commented:
@gsarabandoTT reserved a WH 6U Galaxy to run the installation steps and tests. We kept track of steps verified here: https://docs.google.com/document/d/1VhEfYm66jRW4xBYiPmLMiXEOp95rmcPMf1pdlQzqKSg/edit?tab=t.0#heading=h.rkcxwyi1f35l

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1

gsarabandoTT committed:
Trim trailing whitespace

aliuTT commented on docs/source/tech_reports/WH_Galaxy/Installing_Troubleshooting_Galaxy_WH_6U.md:
```diff
@@ -0,0 +1,196 @@
+
+| Device               | OS              | Python   | Driver (TT-KMD)    | Firmware (TT-Flash)                        | TT-SMI                | TT-Topology                    |
+|----------------------|-----------------|----------|--------------------|--------------------------------------------|-----------------------|--------------------------------|
+| Galaxy (Wormhole 4U) | Ubuntu 22.04    | 3.10     | v1.33 or above     | fw_pack-80.10.1.0                          | v2.2.3 or lower       | v1.1.3, `mesh` config          |
```
Very concerned with having two copies of supported versions, or test cmds to maintain. I would like to see this doc with a focus on "troubleshooting" and link to main install guide for supported versions.

bbeggsTT replied:
Link to INSTALLING.md added instead of installation steps themselves.

bbeggsTT replied:
Link to INSTALLING.md added instead of installation steps themselves.

aliuTT commented on docs/source/tech_reports/WH_Galaxy/Installing_Troubleshooting_Galaxy_WH_6U.md:
```diff
@@ -0,0 +1,196 @@
+
+2. Invoke Metalium Build Scripts - Run the following command to invoke Metalium build scripts:
+
+   `./build_metal.sh`
```
You are making a troubleshooting sequence with tests, might make sense to add the ` --build-test` build option

bbeggsTT replied:
Link to INSTALLING.md added instead of installation steps themselves.

aliuTT commented on docs/source/tech_reports/WH_Galaxy/Installing_Troubleshooting_Galaxy_WH_6U.md:
```diff
@@ -0,0 +1,196 @@
+## TT-SMI Reset and System Information Dump
+To perform a reset on a 6U system run the following command:
+
+`tt-smi --ubb_reset`
```
might get out of date, please link to official tt-smi repo or at least one Metal doc

bbeggsTT committed:
Update and rename Installing_Troubleshooting_Galaxy_WH_6U.md to Galaxy_WH_6U_SW_Guide.md

Updates according to Allan and Raymond:
- Linked to installation guide instead of including specific install steps here.
- Test suite info added.
- TT-SMI command omitted, link to TT-SMI repo added instead.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1

tt-rkim reviewed (dismissed):
Brian - can I ask what's happening here?

You resolved all our conversations without replying to any of them...

From Brian: Allan suggested instead of have the install steps here we link to INSTALLING.md. Since those steps are no longer in this document I resolved the convos. I will make notes there.

tt-rkim commented on docs/source/tech_reports/WH_Galaxy/Galaxy_WH_6U_SW_Guide.md:
```diff
@@ -0,0 +1,111 @@
+
+![](https://github.com/tenstorrent/tt-metal/blob/main/docs/source/common/images/Ethernet_Link_Latency_Test.png)
+
+# FAQ
```
Consider linking to these

https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/1084129418/Policies+for+interfacing+with+developers+and+other+teams
https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/1074659406/TG+6U+Metal+infra+Galaxy+notes+FAQ

and reduce the copy paste

bbeggsTT replied:
It might be better to have FAQ info included in this document to consolidate troubleshooting info. Also I think GitHub has better visibility than Confluence. If you would rather have the link to Confluence I am fine with that. Let me know what you think.

bbeggsTT replied:
Sounds good. I'll include the link with a note about it being for internal use only.

bbeggsTT commented on docs/source/tech_reports/WH_Galaxy/Installing_Troubleshooting_Galaxy_WH_6U.md:
```diff
@@ -0,0 +1,196 @@
+  sudo ./install_dependencies.sh`
+
+## Install the Driver (TT-KMD)
+1. To install DKMS on Ubuntu run the following command:
```
Link to INSTALLING.md added instead of installation steps themselves.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1

bbeggsTT committed:
Update Galaxy_WH_6U_SW_Guide.md, FAQ section linked instead

FAQ section replaced with Internal section and link to internal doc.

bbeggsTT committed:
Merge branch 'main' into bbeggsTT-patch-1
